internal/characters/kuki/skill.go

```diff
@@ -22,11 +22,18 @@ func init() {
 }
 
 func (c *char) Skill(p map[string]int) action.ActionInfo {
-	//remove some hp
-	if 0.7*(c.HPCurrent/c.MaxHP()) > 0.2 {
-		c.HPCurrent = 0.7 * c.HPCurrent
-	} else if (c.HPCurrent / c.MaxHP()) > 0.2 { //check if below 20%
-		c.HPCurrent = 0.2 * c.MaxHP()
+	// only drain HP when above 20% HP
+	if c.HPCurrent/c.MaxHP() > 0.2 {
+		hpdrain := 0.3 * c.HPCurrent
+		// The HP consumption from using this skill can only bring her to 20% HP.
+		if 0.7*c.HPCurrent/c.MaxHP() <= 0.2 {
```

Comment:
I think this is more readable...

```go
if (c.HPCurrent - hpdrain)/c.MaxHP() <= 0.2 {
   hpdrain = hpCurrent - 0.2*c.MaxHP()
}
```

can make the `0.2` a constant or something `const hpDrainThreshold = 0.2`